muxers/yamux/src/lib.rs

```diff
@@ -22,29 +22,27 @@
 
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 
-use futures::{future, prelude::*, ready, stream::BoxStream};
+use futures::{future, prelude::*, ready};
 use libp2p_core::muxing::{StreamMuxer, StreamMuxerEvent};
 use libp2p_core::upgrade::{InboundUpgrade, OutboundUpgrade, UpgradeInfo};
 use std::collections::VecDeque;
 use std::io::{IoSlice, IoSliceMut};
 use std::task::Waker;
 use std::{
-    fmt, io, iter, mem,
+    io, iter,
     pin::Pin,
     task::{Context, Poll},
 };
 use thiserror::Error;
 use yamux::ConnectionError;
 
 /// A Yamux connection.
-pub struct Muxer<C> {
-    /// The [`futures::stream::Stream`] of incoming substreams.
-    incoming: BoxStream<'static, Result<yamux::Stream, yamux::ConnectionError>>,
-    /// Handle to control the connection.
-    control: yamux::Control,
+#[derive(Debug)]
+pub struct Muxer<S> {
+    connection: yamux::Connection<S>,
```

Comment:
@thomaseizinger why rename the type parameter from `C` (I am assuming *Connection*) to `S` (I am assuming *Stream*)?

Reply:
Not sure anymore :)

Don't really care!

Reply:
```suggestion
pub struct Muxer<C> {
    connection: yamux::Connection<C>,
```

Changing to `C` to be consistent with `impl`s below.